src/Sarif.Viewer.VisualStudio.Core/ErrorList/SarifTableDataSource.cs

```diff
@@ -107,6 +106,35 @@ public void AddErrors(IEnumerable<SarifErrorListItem> errors)
             }
         }
 
+        public void UpdateError(int oldItemIdentity, SarifErrorListItem newItem)
+        {
+            if (newItem == null)
+            {
+                return;
+            }
+
+            using (this.tableEntriesLock.EnterWriteLock())
+            {
+                if (this.logFileToTableEntries.TryGetValue(newItem.LogFilePath, out List<SarifResultTableEntry> entries))
+                {
+                    SarifResultTableEntry entryToRemove = entries.First(entry => (int)entry.Identity == oldItemIdentity);
```

Comment:
>First [](http://example.com/codeflow?start=66&length=5)

I think you'll need FirstOrDefault here.

Reply:
Good catch, i wanted to use FirstOrDefault here. Also added a test case for not existing identity.